### PR TITLE
Fix trivia handling in 'use named arguments' refactoring

### LINKED ISSUE #63361: Add argument name refactoring on commented first arg seems to duplicate comment
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Add-argument-name-refactoring-on-comment/10118877)._

---
[severity:Other]
I have this code:

```
static void F(string? x, string? y)
{
    Console.WriteLine($"x: {x??""}");
    Console.WriteLine($"y: {y??""}");
}

F(
        // TODO: 1
        null
        // TODO: 2
    ,   null
    );

```

I run the refactoring "Add argument name 'x'" on line 9 through alt-Enter

It produces the result

```
static void F(string? x, string? y)
{
    Console.WriteLine($"x: {x??""}");
    Console.WriteLine($"y: {y??""}");
}

F(
        // TODO: 1
        x:
        // TODO: 1
        null
        // TODO: 2
    ,   null
    );

```

Note first TODO comment duplicate and x placed in weird place.

---
#### Feedback Bot on 8/12/2022, 01:36 AM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

### LINKED ISSUE #63362: Add argument name refactoring on commented first arg seems to duplicate comment
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Add-argument-name-refactoring-on-comment/10118877)._

---
[severity:Other]
I have this code:

```
static void F(string? x, string? y)
{
    Console.WriteLine($"x: {x??""}");
    Console.WriteLine($"y: {y??""}");
}

F(
        // TODO: 1
        null
        // TODO: 2
    ,   null
    );

```

I run the refactoring "Add argument name 'x'" on line 9 through alt-Enter

It produces the result

```
static void F(string? x, string? y)
{
    Console.WriteLine($"x: {x??""}");
    Console.WriteLine($"y: {y??""}");
}

F(
        // TODO: 1
        x:
        // TODO: 1
        null
        // TODO: 2
    ,   null
    );

```

Note first TODO comment duplicate and x placed in weird place.

---
#### Feedback Bot on 8/12/2022, 01:36 AM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #63361
Fixes #63362

### CODE REVIEW

**Review by jmarolf — APPROVED** [review]

File: src/EditorFeatures/CSharpTest/UseNamedArguments/UseNamedArgumentsTests.cs [hunk 1]
```diff
@@ -552,11 +571,68 @@ public async Task TestNoTrailingArgumentsToName()
...
+            await new VerifyCS.Test
+            {
+                TestCode = initialMarkup,
+                FixedCode = @"class C { void M(int arg1, int arg2, int arg3) => M(1, arg2: 2, arg3: 3); }",
+                ExactActionSetOffered = new[] { string.Format(FeaturesResources.Add_argument_name_0, "arg2") },
+            }.RunAsync();
+        }
+
+        [WorkItem(63362, "https://github.com/dotnet/roslyn/issues/63362")]
+        public async Task TestTrivia()
```
[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/EditorFeatures/CSharpTest/UseNamedArguments/UseNamedArgumentsTests.cs:583]: Yeah I didn't resolve the conflicts properly and lost the attribute. Fixed now.

File: src/Features/CSharp/Portable/UseNamedArguments/CSharpUseNamedArgumentsCodeRefactoringProvider.cs [hunk 2]
```diff
@@ -69,7 +69,7 @@ protected override SeparatedSyntaxList<ArgumentSyntax> GetArguments(BaseArgument
                 => argumentList.WithArguments(SyntaxFactory.SeparatedList(namedArguments, separators));
 
             protected override ArgumentSyntax WithName(ArgumentSyntax argument, string name)
-                => argument.WithNameColon(SyntaxFactory.NameColon(name.ToIdentifierName()));
+                => argument.WithoutTrivia().WithNameColon(SyntaxFactory.NameColon(name.ToIdentifierName()));
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/UseNamedArguments/CSharpUseNamedArgumentsCodeRefactoringProvider.cs:72]: so this feels like it's losing the arg trivia taht actually exists.  How does this work out in practice.  like if there is trailing trivia on the argument, won't this lose that?

why aren't we just moving the leading trivia from teh argument to the new `name-colon`?

Comment (Youssef1313) [reply src/Features/CSharp/Portable/UseNamedArguments/CSharpUseNamedArgumentsCodeRefactoringProvider.cs:72]: @CyrusNajmabadi We shouldn't be losing trivia. The trivia is indeed added to the new node, but by the caller:

https://github.com/dotnet/roslyn/blob/4cc1b4e445f201eb77e452a20b86df42747b49bf/src/Features/Core/Portable/UseNamedArguments/AbstractUseNamedArgumentsCodeRefactoringProvider.cs#L159

Comment (Youssef1313) [reply src/Features/CSharp/Portable/UseNamedArguments/CSharpUseNamedArgumentsCodeRefactoringProvider.cs:72]: I can move the `WithoutTrivia` call to the caller.

Comment (Youssef1313) [reply src/Features/CSharp/Portable/UseNamedArguments/CSharpUseNamedArgumentsCodeRefactoringProvider.cs]: Done in 1697b22.

**Review by CyrusNajmabadi — APPROVED** [review]
Nice!

**Pull request merged**